src/dosext/dpmi/dnative/dnative.c

```diff
@@ -119,9 +122,10 @@ void native_dpmi_set_fpu_state(const emu_fpstate *fpstate)
 #else
     /* i386: convert fxsave state to fsave state */
     convert_from_fxsr(scp_fpregs, fpstate);
-    if ((scp_fpregs->status >> 16) != EMU_X86_FXSR_MAGIC)
-      return;
-    fpregs = &scp_fpregs->status + 1;
+    if ((scp_fpregs->status >> 16) == EMU_X86_FXSR_MAGIC)
+      fpregs = &scp_fpregs->status + 1;
```

Comment:
What is scp_fpregs?
Some cached pointer do sigcontext
on a signal stack? I guess with the
local holder you don't need any
scp_fpregs?

Reply:
yes I changed this to a local holder now in f9c3281.